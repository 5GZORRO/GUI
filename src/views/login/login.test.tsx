/* eslint-disable react/display-name */
/* eslint-disable no-undef */
// @ts-nocheck
// @jest-environment jsdom
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import TestRenderer from 'react-test-renderer'
import Login from './index'

jest.mock('react-text-mask', () => (props: JSX.IntrinsicAttributes & React.ClassAttributes<HTMLInputElement> & React.InputHTMLAttributes<HTMLInputElement>) =>
  <input
    type='text'
    name='key'
    onChange={props.onChange}
    value={props.value}
    onBlur={props.onBlur}
    data-testid={'key'}
  />
)

const mockLogin = jest.fn((key, file) => {
  return Promise.resolve({ key, file })
})

describe('Login', function () {
  it('Snapshot ', () => {
    const tree = TestRenderer.create(
        <Login />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should display correct error message key', () => {
    const { getByTestId, findByText } = render(<Login />)

    getByTestId('submit')

    fireEvent.click(getByTestId('submit'))

    expect(findByText('Please enter a valid key')).toBeTruthy()
    expect(mockLogin).not.toBeCalled()
  })

  it('should display correct error message file', () => {
    const { getByTestId, findByText } = render(<Login />)

    getByTestId('submit')

    fireEvent.click(getByTestId('submit'))

    expect(findByText('Please enter a file')).toBeTruthy()
    expect(mockLogin).not.toBeCalled()
  })

  it.skip('should display matching error when key is invalid', async () => {
    const { getByRole, getByTestId } = render(<Login />)

    fireEvent.input(getByTestId('key'), {
      target: {
        value: '123 123 123'
      }
    })

    fireEvent.input(getByRole('textbox', 'file-name'), {
      target: {
        value: [{
          lastModified: 1613853259704,
          lastModifiedDate: 'Sat Feb 20 2021 20:34:19',
          name: 'test.pdf',
          size: 211571,
          type: 'application/pdf',
          webkitRelativePath: ''
        }]
      }
    })

    fireEvent.submit(getByTestId('submit'))
    console.log(getByTestId('key').value)
    // expect(getByTestId('key').value).toBe('')
    // expect(getByRole('textbox', 'file-name').value).toBe('')
  })
})
