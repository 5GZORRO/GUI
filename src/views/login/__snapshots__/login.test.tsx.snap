// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot  1`] = `
<div
  className="c-app c-default-layout flex-row align-items-center"
>
  <div
    className="container"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="justify-content-center row"
      >
        <div
          className="d-flex justify-content-center align-items-center mb-5 col-5"
        >
          <svg
            fill="none"
            height={120}
            viewBox="0 0 215 120"
            width={215}
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clipPath="url(#prefix__clip0)"
            >
              <path
                d="M65.86 86.611h15.496v5.399h-9.744v7.335c.998-.118 1.996-.118 3.053-.118 2.465 0 4.343.646 5.635 1.878 1.291 1.232 1.937 3.051 1.937 5.457v5.692c0 2.406-.822 4.284-2.407 5.634-1.585 1.349-4.05 1.995-7.337 1.995-2.29 0-4.579-.352-6.868-1.057v-5.574c2.407.821 4.637 1.232 6.692 1.232 2.759 0 4.109-.997 4.109-2.934v-4.342c0-.998-.235-1.761-.764-2.23-.47-.469-1.408-.645-2.7-.645-.41 0-2.758.176-7.16.469v-18.19h.058zm20.133 7.57c0-2.465.881-4.342 2.642-5.75 1.76-1.409 4.109-2.113 7.102-2.113 2.994 0 5.342.293 7.044.88v5.457c-1.937-.704-4.05-1.056-6.34-1.056-3.052 0-4.636 1.056-4.636 3.11v16.9c0 1.936 1.291 2.934 3.874 2.934 1.115 0 1.995-.235 2.7-.646v-11.56h5.694v15.375c-2.524 1.29-5.283 1.995-8.394 1.995-3.111 0-5.518-.646-7.162-1.937-1.643-1.291-2.465-3.169-2.465-5.633V94.181h-.059zm21.66-7.57h17.082v4.695l-11.27 23.295h11.328v4.93h-17.374v-4.93l11.035-23.12h-10.801v-4.87zm20.427 7.746c0-2.523.764-4.518 2.348-5.985 1.585-1.467 3.875-2.23 6.986-2.23 3.052 0 5.4.763 6.985 2.23s2.406 3.462 2.406 5.985v17.428c0 2.523-.821 4.518-2.406 5.985-1.585 1.467-3.933 2.23-6.985 2.23-3.053 0-5.401-.763-6.986-2.23-1.584-1.467-2.348-3.462-2.348-5.985V94.357zm12.914.235c0-2.171-1.174-3.286-3.58-3.286-2.407 0-3.581 1.115-3.581 3.286v16.899c0 2.172 1.174 3.287 3.581 3.287 2.406 0 3.58-1.115 3.58-3.287v-16.9zm15.732 12.557v12.382h-5.753v-32.92h9.744c2.583 0 4.637.646 6.046 1.878 1.409 1.291 2.113 3.051 2.113 5.398v5.634c0 3.697-1.35 6.044-4.05 7.041l5.165 12.969h-6.339l-4.579-12.382h-2.347zm0-15.609v10.915h3.345c1.057 0 1.82-.235 2.29-.704.469-.47.704-1.233.704-2.289v-5.164c0-1.877-.998-2.816-2.994-2.816h-3.345v.058zm22.599 15.609v12.382h-5.753v-32.92h9.744c2.583 0 4.637.646 6.046 1.878 1.409 1.291 2.113 3.051 2.113 5.398v5.634c0 3.697-1.35 6.044-4.05 7.041l5.165 12.969h-6.339l-4.579-12.382h-2.347zm0-15.609v10.915h3.345c1.057 0 1.82-.235 2.29-.704.469-.47.704-1.233.704-2.289v-5.164c0-1.877-.998-2.816-2.994-2.816h-3.345v.058zm15.966 2.817c0-2.523.763-4.518 2.348-5.985 1.584-1.467 3.874-2.23 6.985-2.23s5.4.763 6.985 2.23 2.407 3.462 2.407 5.985v17.428c0 2.523-.822 4.518-2.407 5.985-1.585 1.467-3.933 2.23-6.985 2.23-3.053 0-5.401-.763-6.985-2.23-1.585-1.467-2.348-3.462-2.348-5.985V94.357zm12.972.235c0-2.171-1.174-3.286-3.581-3.286-2.406 0-3.58 1.115-3.58 3.286v16.899c0 2.172 1.174 3.287 3.58 3.287 2.407 0 3.581-1.115 3.581-3.287v-16.9z"
                fill="#fff"
              />
              <path
                d="M120.802 36.968l-2.876-2.582-4.285 3.873 3.346 2.993c.88.763 1.35 1.819 1.35 2.875s-.47 2.054-1.35 2.875L98.497 63.61c-1.938 1.76-5.283 1.76-7.22 0l-18.49-16.665c-.881-.763-1.35-1.82-1.35-2.876 0-1.056.469-2.053 1.35-2.875l4.284-3.873-4.284-3.873-3.816 3.462c-2.113 1.878-3.228 4.401-3.228 7.1 0 2.7 1.174 5.223 3.228 7.101l18.49 16.606c2.113 1.878 4.755 2.817 7.455 2.817s5.4-.939 7.455-2.817l18.49-16.606c2.113-1.878 3.228-4.401 3.228-7.1 0-2.64-1.174-5.164-3.287-7.042z"
                fill="url(#prefix__paint0_linear)"
              />
              <path
                d="M91.277 24.587c1.937-1.76 5.283-1.76 7.22 0l7.161 6.396 4.285-3.873-7.572-6.807c-4.168-3.755-10.742-3.755-14.91 0l-6.28 5.633 4.285 3.873 5.81-5.222z"
                fill="url(#prefix__paint1_linear)"
              />
              <path
                d="M177.035 44.069c0-2.29-.939-4.402-2.7-5.986l-14.968-13.496c-3.522-3.169-8.981-3.169-12.503 0l-4.285 3.814 3.933 3.52 3.874-3.461c1.468-1.35 3.992-1.35 5.459 0l14.968 13.496c.646.587.998 1.35.998 2.112 0 .763-.352 1.526-.998 2.113l-14.968 13.496c-1.467 1.35-3.991 1.35-5.459 0l-14.968-13.496c-.646-.587-.998-1.35-.998-2.112 0-.763.352-1.526.998-2.113l3.522-3.169-3.933-3.52-3.111 2.757c-1.761 1.585-2.7 3.697-2.7 5.986 0 2.288.939 4.4 2.7 5.985l14.968 13.496c1.761 1.585 3.992 2.348 6.222 2.348 2.231 0 4.52-.763 6.222-2.348l14.968-13.496c1.82-1.526 2.759-3.638 2.759-5.926z"
                fill="url(#prefix__paint2_linear)"
              />
              <path
                d="M27.647 31.511c-6.926 0-12.62 5.634-12.62 12.616 0 6.925 5.635 12.617 12.62 12.617 6.927 0 12.62-5.634 12.62-12.617 0-6.982-5.693-12.616-12.62-12.616zm0 19.717a7.088 7.088 0 01-7.102-7.1c0-3.932 3.17-7.16 7.102-7.16a7.088 7.088 0 017.103 7.1c0 3.932-3.17 7.16-7.103 7.16zm58.405-17.37L55.94 6.809C48.368 0 37.45-1.878 28.117 1.994 11.035 9.095 0 25.585 0 44.07c0 18.484 11.035 34.973 28.117 42.073a25.96 25.96 0 0010.037 1.995c6.457 0 12.855-2.347 17.786-6.806l21.19-19.072-4.52-4.048-20.72 18.66c-5.87 5.281-14.323 6.748-21.543 3.755-14.792-6.161-24.36-20.479-24.36-36.557A39.521 39.521 0 0130.347 7.51c7.22-2.992 15.673-1.525 21.543 3.756l30.112 27.051c1.703 1.526 2.7 3.58 2.7 5.75 0 2.172-.939 4.167-2.7 5.752l-.939.88 4.52 4.049.47-.47c2.993-2.7 4.695-6.337 4.695-10.21-.058-3.873-1.702-7.511-4.696-10.21z"
                fill="url(#prefix__paint3_linear)"
              />
              <path
                d="M149.212 37.555l-16.611-14.963c-3.816-3.462-9.803-3.462-13.677 0l-16.671 14.963c-1.937 1.702-2.993 4.049-2.993 6.514 0 2.464 1.056 4.811 2.993 6.513l3.464 3.11 4.109-3.697-3.875-3.52c-.763-.646-1.174-1.526-1.174-2.465 0-.88.411-1.76 1.174-2.465l16.612-14.963c1.702-1.526 4.579-1.526 6.281 0l16.67 15.08c.763.646 1.174 1.526 1.174 2.465 0 .88-.411 1.76-1.174 2.465l-2.993 2.699 4.109 3.697 2.582-2.347c1.937-1.702 2.994-4.05 2.994-6.514-.059-2.523-1.115-4.812-2.994-6.572z"
                fill="url(#prefix__paint4_linear)"
              />
              <path
                d="M128.903 61.496c-1.703 1.526-4.579 1.526-6.281 0l-4.755-4.283-4.109 3.697 5.166 4.635a10.27 10.27 0 006.809 2.582c2.465 0 4.931-.88 6.809-2.582l6.457-5.81-4.109-3.696-5.987 5.457z"
                fill="url(#prefix__paint5_linear)"
              />
            </g>
            <defs>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="prefix__paint0_linear"
                x1={65.668}
                x2={124.07}
                y1={52.094}
                y2={52.094}
              >
                <stop
                  stopColor="#26005F"
                />
                <stop
                  offset={0.136}
                  stopColor="#2F005F"
                />
                <stop
                  offset={0.356}
                  stopColor="#47005F"
                />
                <stop
                  offset={0.635}
                  stopColor="#6F005F"
                />
                <stop
                  offset={0.959}
                  stopColor="#A5005F"
                />
                <stop
                  offset={1}
                  stopColor="#AD005F"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="prefix__paint1_linear"
                x1={81.148}
                x2={109.923}
                y1={24.269}
                y2={24.269}
              >
                <stop
                  stopColor="#26005F"
                />
                <stop
                  offset={0.136}
                  stopColor="#2F005F"
                />
                <stop
                  offset={0.356}
                  stopColor="#47005F"
                />
                <stop
                  offset={0.635}
                  stopColor="#6F005F"
                />
                <stop
                  offset={0.959}
                  stopColor="#A5005F"
                />
                <stop
                  offset={1}
                  stopColor="#AD005F"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="prefix__paint2_linear"
                x1={129.162}
                x2={177.042}
                y1={44.089}
                y2={44.089}
              >
                <stop
                  stopColor="#004A93"
                />
                <stop
                  offset={0.067}
                  stopColor="#0C558C"
                />
                <stop
                  offset={0.381}
                  stopColor="#43856B"
                />
                <stop
                  offset={0.653}
                  stopColor="#6BA853"
                />
                <stop
                  offset={0.868}
                  stopColor="#83BE44"
                />
                <stop
                  offset={1}
                  stopColor="#8CC63F"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="prefix__paint3_linear"
                x1={0}
                x2={90.706}
                y1={44.089}
                y2={44.089}
              >
                <stop
                  stopColor="#ED1C24"
                />
                <stop
                  offset={0.304}
                  stopColor="#F14A22"
                />
                <stop
                  offset={0.609}
                  stopColor="#F47220"
                />
                <stop
                  offset={0.851}
                  stopColor="#F68A1E"
                />
                <stop
                  offset={1}
                  stopColor="#F7931E"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="prefix__paint4_linear"
                x1={99.294}
                x2={152.163}
                y1={36.869}
                y2={36.869}
              >
                <stop
                  stopColor="#10015D"
                />
                <stop
                  offset={1}
                  stopColor="#004A93"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="prefix__paint5_linear"
                x1={113.733}
                x2={138.984}
                y1={62.134}
                y2={62.134}
              >
                <stop
                  stopColor="#10015D"
                />
                <stop
                  offset={1}
                  stopColor="#004A93"
                />
              </linearGradient>
              <clipPath
                id="prefix__clip0"
              >
                <path
                  d="M0 0h214.016v120H0z"
                  fill="#fff"
                />
              </clipPath>
            </defs>
          </svg>
        </div>
      </div>
      <div
        className="justify-content-center row"
      >
        <div
          className="d-flex justify-content-center align-items-center col-5"
        >
          <div
            className="p-5 w-100 card card-accent-#0403"
          >
            <div
              className="p-0 card-body"
            >
              <h1
                className="mb-4"
              >
                Login
              </h1>
              <p
                className="text-muted"
              >
                Sign In to your account
              </p>
              <div
                className="form-group mb-4"
              >
                <label
                  className=""
                >
                  Enter Key
                </label>
                <div
                  className="input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <div
                      className="input-group-text"
                    >
                      <svg
                        height={18}
                        viewBox="0 0 18 22"
                        width={22}
                      >
                        <path
                          d="M18 10C18 15.55 14.16 20.74 9 22C3.84 20.74 0 15.55 0 10V4L9 0L18 4V10ZM9 20C12.75 19 16 14.54 16 10.22V5.3L9 2.18L2 5.3V10.22C2 14.54 5.25 19 9 20ZM9 5C9.79565 5 10.5587 5.31607 11.1213 5.87868C11.6839 6.44129 12 7.20435 12 8C12 9.31 11.17 10.42 10 10.83V13H12V15H10V17H8V10.83C6.83 10.42 6 9.31 6 8C6 7.20435 6.31607 6.44129 6.87868 5.87868C7.44129 5.31607 8.20435 5 9 5ZM9 7C8.73478 7 8.48043 7.10536 8.29289 7.29289C8.10536 7.48043 8 7.73478 8 8C8 8.26522 8.10536 8.51957 8.29289 8.70711C8.48043 8.89464 8.73478 9 9 9C9.26522 9 9.51957 8.89464 9.70711 8.70711C9.89464 8.51957 10 8.26522 10 8C10 7.73478 9.89464 7.48043 9.70711 7.29289C9.51957 7.10536 9.26522 7 9 7Z"
                          fill="#6C6E7E"
                        />
                      </svg>
                    </div>
                  </div>
                  <input
                    data-testid="key"
                    name="key"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <small
                  className="form-text text-muted mt-2"
                >
                  ex. 999 999 999 9999
                </small>
              </div>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="file-input"
                >
                  Certified upload
                </label>
                <div
                  className="m-0 d-flex align-items-center row"
                  style={
                    Object {
                      "borderRadius": "4px",
                      "boxShadow": "0px 4px 4px rgba(0, 0, 0, 0.25)",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col-sm-7 col-lg-8"
                  >
                    <div
                      className="p-2"
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#181924",
                          "borderRadius": "4px",
                          "display": "flex",
                          "height": "36px",
                          "width": "36px",
                        }
                      }
                    >
                      <svg
                        height={18}
                        viewBox="0 0 16 20"
                        width={18}
                      >
                        <path
                          d="M10 0l6 6v12a2 2 0 01-2 2H2a2 2 0 01-2-2V2a2 2 0 012-2h8zm4 18V7H9V2H2v16h12zm-7.08-7.69c-.24-.77-.77-3.23.63-3.27 1.4-.04.48 3.12.48 3.12.39 1.49 2.02 2.56 2.02 2.56.5-.15 3.35-.48 2.95 1-.43 1.48-3.5.09-3.5.09-1.95.14-3.41.66-3.41.66-1.13 2.11-2.45 3.03-2.99 2.14-.67-1.11 2.13-2.54 2.13-2.54 1.45-2.35 1.67-3.72 1.69-3.76zm.65.84c-.4 1.3-1.2 2.69-1.2 2.69.85-.34 2.71-.73 2.71-.73-1.14-1-1.49-1.95-1.51-1.96zm3.14 2.17s1.75.65 1.79.39c.07-.27-1.33-.51-1.79-.39zm-5.66 1.49c-.77.3-1.51 1.58-1.33 1.58.18.01.91-.6 1.33-1.58zm2.52-5.55c0-.05.43-1.68 0-1.73-.3-.03-.01 1.69 0 1.73z"
                          fill="#6C6E7E"
                        />
                      </svg>
                    </div>
                    <div
                      className="p-2"
                    >
                      <p
                        style={
                          Object {
                            "margin": "0 12px",
                            "overflow": "hidden",
                            "textOverflow": "ellipsis",
                            "whiteSpace": "nowrap",
                            "width": "200px",
                          }
                        }
                      >
                        Choose File
                      </p>
                      <p
                        style={
                          Object {
                            "color": "#8A93A2",
                            "margin": "0 12px",
                          }
                        }
                      >
                        Type File: 
                        <strong>
                          PDF
                        </strong>
                      </p>
                    </div>
                  </div>
                  <div
                    className="d-flex justify-content-end align-items-center col-sm-5 col-lg-4"
                  >
                    <label
                      className="custom-file-upload m-0"
                    >
                      <input
                        accept=".pdf"
                        className="custom-file-input"
                        id="file-input"
                        name="file-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        style={
                          Object {
                            "border": "1px solid #ccc",
                            "cursor": "pointer",
                            "display": "inline-block",
                            "padding": "6px 12px",
                            "width": "0px",
                          }
                        }
                        type="file"
                      />
                      <span
                        style={
                          Object {
                            "color": "#9DA5B1",
                            "cursor": "pointer",
                            "textDecorationLine": "underline",
                          }
                        }
                      >
                        <strong>
                          Choose File
                        </strong>
                      </span>
                    </label>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="text-right mb-4 col-12"
                >
                  <p
                    className="px-0 text-uppercase font-12 text-light cursor-pointer"
                    onClick={[Function]}
                  >
                    <u>
                      did you have any problem?
                    </u>
                  </p>
                </div>
                <div
                  className="col-12"
                >
                  <button
                    className="px-5 text-uppercase btn btn-gradient"
                    data-testid="submit"
                    onClick={[Function]}
                    type="submit"
                  >
                    submit
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <div
      className="justify-content-center row"
    >
      <div
        className="d-flex justify-content-center align-items-center w-63 col-5"
      >
        <button
          className="btn btn-secondary btn-lg"
          onClick={[Function]}
          type="button"
        >
          Create new  Account
        </button>
      </div>
    </div>
  </div>
</div>
`;
